### PR TITLE
Site Logo: add padding and margin support

### PR DESCRIPTION
**Pull request by colorful-tones**
Related:

- https://github.com/WordPress/gutenberg/issues/43241
- https://github.com/WordPress/gutenberg/issues/43243

## What?

Add padding and margin support to the Site Logo block.

## Why?

To create consistency across blocks.

## How?

Added the relevant block support in `block.json` and updated Core Block Reference docs.

## Testing Instructions

1. Go to Appearance > Editor (beta)
2. Add the Site Logo block to the template
3. Configure margin and padding and verify the output.

site-logo-supports

### CODE REVIEW

File: docs/reference-guides/core-blocks.md
Comment (ndiego) [inline docs/reference-guides/core-blocks.md]: This is minor, but can you alphabetize the active supports followed by the alphabetized excluded supports? So spacing would come after color.

Comment (colorful-tones) [reply docs/reference-guides/core-blocks.md]: Nothing too minor. 👍 I addressed this and alpha sorted. Thanks!

**Comment by carolinan** [comment]
For this block, I was not able to get top and bottom margin to work.

**Comment by colorful-tones** [comment]
> For this block, I was not able to get top and bottom margin to work.

@carolinan I pulled in the latest from trunk and tested again and all `margin` and `padding` axes are working. 🤔 

Here is a video:


https://user-images.githubusercontent.com/405912/187998358-f37ac1f8-e8e4-4178-ae6a-c3a6a9342b4b.mp4

My local testing environment:

- WordPress 6.0.2
- Twenty Twenty-Two 1.2

**Comment by carolinan** [comment]
theme.json only, empty theme, logo is not nested:
Screen Shot 2022-09-02 at 03 59 27

theme.json only, TT2, logo in the default position **inside the nested header**, front view
Screen Shot 2022-09-02 at 04 02 10

global styles, padding and margin set to medium spacing preset, TT2:
Screen Shot 2022-09-02 at 04 05 09

global styles, padding and margin set to medium spacing preset, TT2, site logo **not nested** (its between the site header and main query)
Screen Shot 2022-09-02 at 04 08 43

**Comment by carolinan** [comment]
Global styles, spacing preset 80, logo is not nested, Twenty Twenty-Three:
Screen Shot 2022-09-02 at 04 25 02

**Comment by colorful-tones** [comment]
Thanks for the additional testing @carolinan 

It took me a bit to decipher whether your testing was offering successful or failure in moving this issue along, but I see that there needs to be additional CSS cascade resolution for this to work. 

The `.wp-blocks > * + * {}` seems to be the culprit.

**Review by carolinan — APPROVED** [review]
These margin issues has to be bigger than these two blocks, and not caused by the PR.
My concern is that these changes will go into 6.1 without the margin begin fixed, and it may appear broken to many users...

**Comment by colorful-tones** [comment]
> These margin issues has to be bigger than these two blocks, and not caused by the PR.

👍 

> My concern is that these changes will go into 6.1 without the margin begin fixed, and it may appear broken to many users...

I share these concerns as well and I'm glad you are keeping them on the radar. Do you happen to know if there is an existing issue related to the over-specificity for margin @carolinan ?

**Comment by aaronrobertshaw** [comment]
There are a few issues already around the topic of layout and margin styles conflicting:
- https://github.com/WordPress/gutenberg/issues/43404
- https://github.com/Automattic/wp-calypso/issues/64438

This has also been raised on some recent PRs adding margin support to blocks:
- https://github.com/WordPress/gutenberg/pull/43365#issuecomment-1220093440
- https://github.com/WordPress/gutenberg/pull/43366#issuecomment-1220112028
- https://github.com/WordPress/gutenberg/pull/43402#issuecomment-1220249799

**Review by andrewserong — APPROVED** [review]
Thanks for opening a PR for this one @colorful-tones, this is testing nicely for me, too. It looks like the margin styling conflicts issue is the one that @aaronrobertshaw linked to in #43404.

From my perspective, we shouldn't let that issue block merging in these opt-in PRs, as we will still have time to investigate bugs in style output after the 6.1 feature freeze. Dealing with specificity in the layout support and top margins is quite tricky to untangle as sometimes folks will want the Layout support to take precedence, and other times the block-level setting in `theme.json`. Since the margin support does support block-level in post content overriding Layout styles, in most cases, folks should be able to still come up with manual workarounds, so I don't think it's a blocker for these opt-ins.

LGTM!

**Comment by andrewserong** [comment]
@colorful-tones one other thing that would be good to add before landing this, is the `box-sizing: border-box` rule like you added in https://github.com/WordPress/gutenberg/pull/43519 (though we can always add it after the fact). Most of the time I think site logos will typically be included in a Row block where this isn't so noticeable, but if we have this block sitting next to a Group block with padding, then it's noticeable that the padding behaves differently in TwentyTwentyTwo:

image

**Comment by colorful-tones** [comment]
> one other thing that would be good to add before landing this, is the box-sizing: border-box rule like you added in https://github.com/WordPress/gutenberg/pull/43519

@andrewserong good call! I added the `box-sizing: border-box` for this block as well. Thanks for the testing and feedback. ❤️

**Review by andrewserong — APPROVED** [review]
Thanks for updating @colorful-tones, this is working nicely for me now! I'll merge this in and update the tracking issue 👍

**Pull request merged**
